galite-tests/src/test/kotlin/org/kopi/galite/demo/stock/StockR.kt

```diff
@@ -0,0 +1,134 @@
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1 as published by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package org.kopi.galite.demo.stock
+
+import java.util.Locale
+
+import org.jetbrains.exposed.sql.selectAll
+import org.jetbrains.exposed.sql.transactions.transaction
+
+import org.kopi.galite.demo.Stock
+import org.kopi.galite.domain.Domain
+import org.kopi.galite.form.dsl.Key
+import org.kopi.galite.report.FieldAlignment
+import org.kopi.galite.report.Report
+import org.kopi.galite.report.VReport
+
+/**
+ * STock Report
+ */
+object StockR : Report() {
+  override val locale = Locale.FRANCE
+
+  override val title = "Stocks_Report"
+
+  val action = menu("Action")
+
+  val csv = actor(
+          ident = "CSV",
+          menu = action,
+          label = "CSV",
+          help = "Obtenir le format CSV",
+  ) {
+    key = Key.F8          // key is optional here
+    icon = "export"  // icon is optional here
+  }
+
+  val xls = actor(
+          ident = "XLS",
+          menu = action,
+          label = "XLS",
+          help = "Obtenir le format Excel (XLS)",
+  ) {
+    key = Key.SHIFT_F8          // key is optional here
+    icon = "export"  // icon is optional here
+  }
+
+  val xlsx = actor(
+          ident = "XLSX",
+          menu = action,
+          label = "XLSX",
+          help = "Obtenir le format Excel (XLSX)",
+  ) {
+    key = Key.SHIFT_F8          // key is optional here
+    icon = "export"  // icon is optional here
+  }
+
+  val pdf = actor(
+          ident = "PDF",
+          menu = action,
+          label = "PDF",
+          help = "Obtenir le format PDF",
+  ) {
+    key = Key.F9          // key is optional here
+    icon = "export"  // icon is optional here
+  }
+
+  val cmdCSV = command(item = csv) {
+    action = {
+      model.export(VReport.TYP_CSV)
+    }
+  }
+
+  val cmdPDF = command(item = pdf) {
+    action = {
+      model.export(VReport.TYP_PDF)
+    }
+  }
+
+  val cmdXLS = command(item = xls) {
+    action = {
+      model.export(VReport.TYP_XLS)
+    }
+  }
+
+  val cmdXLSX = command(item = xlsx) {
+    action = {
+      model.export(VReport.TYP_XLSX)
+    }
+  }
+
+  val idStckPdt = field(Domain<Int>(25)) {
+    label = "Product_ID"
+    help = "The product ID"
+    align = FieldAlignment.LEFT
+  }
+  val idStckProv = field(Domain<Int>(25)) {
+    label = "Provider_ID"
+    help = "The provider id"
+    align = FieldAlignment.LEFT
+  }
```

Comment:
id display has no meaning , we need to show product name + provider

Reply:
in one stock we can have more than one product, that's why we need to specify which product we are talking about